src/main/java/io/cdap/plugin/servicenow/sink/service/ServiceNowSinkAPIRequestImpl.java

```diff
@@ -0,0 +1,262 @@
+          if (failedRequestId != null) {
+            unservicedRequestsArray.add(failedRequestId);
+          }
+
+          // Process unserviced requests array into unserviced requests map
+          Map<String, RestRequest> unservicedRequestsMap = processUnservicedRequestsArray(restRequestsMap,
+                                                                                          unservicedRequestsArray);
+          // Retry unserviced requests
+          createPostRequest(unservicedRequestsMap, accessToken);
+        }
+      }
+    } catch (IOException e) {
+      LOG.error("Error while connecting to ServiceNow", e.getMessage());
+      throw new RetryableException();
+    } catch (Exception e) {
+      throw e;
+    }
+  }
+
+  private String getErrorMessage(String responseBody) {
+    try {
+      JsonObject jo = gson.fromJson(responseBody, JsonObject.class);
+      return jo.getAsJsonObject(ServiceNowConstants.ERROR).get(ServiceNowConstants.MESSAGE).getAsString();
+    } catch (Exception e) {
+      return e.getMessage();
+    }
+  }
+
+  private ServiceNowBatchRequest getPayloadRequest(Map<String, RestRequest> restRequests) {
+    ServiceNowBatchRequest payloadRequest = new ServiceNowBatchRequest();
+    payloadRequest.setBatchRequestId(UUID.randomUUID().toString());
+    payloadRequest.setRestRequests(new ArrayList<>(restRequests.values()));
+    return payloadRequest;
+  }
+
+  /**
+   * Process the unserviced requests array into an unserviced requests map.    .
+   *
+   * @param restRequestsMap         The map of rest requests
+   * @param unservicedRequestsArray An array of unserviced requests
+   * @return unservicedRequestsMap The map of unserviced requests
+   */
+  private Map<String, RestRequest> processUnservicedRequestsArray(Map<String, RestRequest> restRequestsMap,
+                                                                  JsonArray unservicedRequestsArray) {
+    Map<String, RestRequest> unservicedRequestsMap = new HashMap<>();
+    for (int i = 0; i < unservicedRequestsArray.size(); i++) {
+      String requestId = unservicedRequestsArray.get(i).getAsString();
+      RestRequest request = restRequestsMap.get(requestId);
+      if (request == null) {
+        LOG.error("No Rest Request found for Request Id {}", requestId);
+      }
+      unservicedRequestsMap.put(requestId, request);
+    }
+    return unservicedRequestsMap;
+  }
+
+  /**
+   * Retries to insert/update the records into ServiceNow table when RetryableException is thrown          .
+   *
+   * @param restRequestsMap The map of rest Requests
+   */
+  public void createPostRequestRetryableMode(Map<String, RestRequest> restRequestsMap) throws ExecutionException,
+    RetryException {
+    String accessToken = restApi.getAccessTokenRetryableMode();
+    Callable<Boolean> fetchRecords = () -> {
+      createPostRequest(restRequestsMap, accessToken);
+      return true;
+    };
+
+    Retryer<Boolean> retryer = RetryerBuilder.<Boolean>newBuilder()
```

Comment:
if you're not doing anything with the return value you can just make it a Retryer<Void>

Reply:
Replaced `Retryer<Boolean>` with `Retryer` e8b33614f910fa16f06ffcf185ed0d3a23d9fda1